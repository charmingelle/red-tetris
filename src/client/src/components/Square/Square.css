.square {
  width: 50px;
  height: 50px;
  outline: 1px solid grey;
  position: absolute;
}
