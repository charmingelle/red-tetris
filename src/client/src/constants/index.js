export const LEFT = 'ArrowLeft';
export const RIGHT = 'ArrowRight';
export const DOWN = 'ArrowDown';
export const UP = 'ArrowUp';

export const UPDATE_MY_ID = 'UPDATE_MY_ID';
export const LOAD_PEOPLE = 'LOAD_PEOPLE';
export const LOAD_ROOMS = 'LOAD_ROOMS';
export const LOAD_ROOM = 'LOAD_ROOM';
export const SET_TETRO = 'SET_TETRO';
export const MOVE_TETRO = 'MOVE_TETRO';
export const MOVE_TETRO_DOWN = 'MOVE_TETRO_DOWN';
export const DROP_TETRO = 'DROP_TETRO';
export const ROTATE_TETR0 = 'ROTATE_TETRO';
export const UPDATE_MY_ROOM_ID = 'UPDATE_MY_ROOM_ID';

export const RIGHT_LIMIT = 10;
export const BOTTOM_LIMIT = 23;
export const INVISIBLE_ROW_AMOUNT = 3;
export const SQUARE_WIDTH = 40;
export const OTHER_SQUARE_WIDTH = 8;
export const WIDTH_IN_SQUARES = 10;
export const HEIGHT_IN_SQUARES = 20;

export const PENALTY_COLOR = {
  main: '#666666',
  lighter: '#999',
  darker: '#333',
};
