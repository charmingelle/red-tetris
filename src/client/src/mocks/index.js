export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
