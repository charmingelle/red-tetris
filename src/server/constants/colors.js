exports.RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

exports.ORANGE = {
  main: '#f77700',
  lighter: '#f7923a',
  darker: '#9b4e00',
};

exports.YELLOW = {
  main: '#cccc00',
  lighter: '#feff00',
  darker: '#9a9800',
};

exports.GREEN = {
  main: '#129c0a',
  lighter: '#3bd652',
  darker: '#077001',
};

exports.BLUE = {
  main: '#06cccc',
  lighter: '#0cffff',
  darker: '#108d8e',
};

exports.PURPLE = {
  main: '#0d05f3',
  lighter: '#3033f2',
  darker: '#020279',
};

exports.VIOLET = {
  main: '#9901cc',
  lighter: '#cc02ff',
  darker: '#660099',
};
