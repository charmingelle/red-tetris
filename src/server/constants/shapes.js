exports.LINE = {
  shape: [
    [0, 0, 1, 0],
    [0, 0, 1, 0],
    [0, 0, 1, 0],
    [0, 0, 1, 0],
  ],
};

exports.GI = {
  shape: [
    [0, 0, 0],
    [1, 1, 1],
    [0, 0, 1],
  ],
  initRow: 1,
};

exports.GI_MIRROR = {
  shape: [
    [0, 0, 0],
    [1, 1, 1],
    [1, 0, 0],
  ],
  initRow: 1,
};

exports.SQUARE = {
  shape: [
    [1, 1],
    [1, 1],
  ],
  initRow: 2,
};

exports.ZI = {
  shape: [
    [0, 0, 0],
    [1, 1, 0],
    [0, 1, 1],
  ],
  initRow: 1,
};

exports.ZI_MIRROR = {
  shape: [
    [0, 0, 0],
    [0, 1, 1],
    [1, 1, 0],
  ],
  initRow: 1,
};

exports.TI = {
  shape: [
    [0, 0, 0],
    [1, 1, 1],
    [0, 1, 0],
  ],
  initRow: 1,
};

exports.PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];
