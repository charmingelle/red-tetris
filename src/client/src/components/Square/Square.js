import React from "react";
import "./Square.css";
import { SHIFT } from "../../constants";

export const Square = ({ left, top, color }) => (
  <div
    className="square"
    style={{
      left: `${left * SHIFT}px`,
      top: `${top * SHIFT}px`,
      backgroundColor: color
    }}
  ></div>
);
