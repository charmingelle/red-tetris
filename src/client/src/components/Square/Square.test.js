import React from 'react';
import { Square } from './Square';
import renderer from 'react-test-renderer';
import { SQUARE_WIDTH } from '../../constants';
import { RED } from '../../mocks';

test('Square', () => {
  expect(
    renderer
      .create(
        <Square
          left={3}
          top={4}
          color={RED}
          width={SQUARE_WIDTH}
          borderWidth={SQUARE_WIDTH / 8}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
