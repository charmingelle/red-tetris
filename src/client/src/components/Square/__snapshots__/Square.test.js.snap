// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square 1`] = `
<div
  className="square"
  style={
    Object {
      "backgroundColor": "#c00",
      "borderBottom": "5px solid #980000",
      "borderLeft": "5px solid #f00",
      "borderRight": "5px solid #980000",
      "borderTop": "5px solid  #f00",
      "height": "40px",
      "left": "120px",
      "top": "160px",
      "width": "40px",
    }
  }
/>
`;
